components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -0,0 +1,154 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+use crate::LocaleCanonicalizer;
+use crate::LocaleCanonicalizerError;
+use icu_locid::subtags;
+use icu_locid::LanguageIdentifier;
+use icu_locid::Locale;
+use icu_provider::prelude::*;
+use icu_provider::structs::likelysubtags::*;
+use std::borrow::Cow;
+
+impl LocaleCanonicalizer {
+    /// A constructor which takes a DataProvider and creates a
+    /// LocaleCanonicalizer.
+    pub fn new(provider: &dyn DataProvider) -> Result<LocaleCanonicalizer, DataError> {
+        let payload: Cow<LikelySubtagsV1> = provider
+            .load(&DataRequest {
+                data_key: key::LIKELY_SUBTAGS_V1,
+                data_entry: DataEntry {
+                    variant: None,
+                    langid: "und".parse().unwrap(),
+                },
+            })?
+            .take_payload()?;
+
+        Ok(LocaleCanonicalizer {
+            likely_subtags: payload.into_owned(),
```

Comment:
Why do you need to `.into_owned()` it?  That voids the benefit of having a Cow.  You can make LocaleCanoicalizer have a lifetime parameter, the same lifetime as the Cow.

Reply:
Oops, that was left over from when I was sorting here. Thanks for catching that!

Reply:
This doesn't appear to be resolved yet.

Reply:
That is embarrassing, I made those changes on my laptop before the holidays and forgot to bring the changes over to my desktop when I came back to work this week.